src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -478,7 +478,6 @@ void VideoDecoder::addVideoStreamDecoder(
       options.colorConversionLibrary.value_or(defaultColorConversionLibrary);
 
   if (colorConversionLibrary == ColorConversionLibrary::FILTERGRAPH) {
-    initializeFilterGraphForStream(streamNumber, options);
```

Comment:
(Not critical for merging)

Seems like lines 478 to 492 can now be collapsed into just a one-liner?

```C
streamInfo.colorConversionLibrary = options.colorConversionLibrary.value_or(defaultColorConversionLibrary);
```



In particular I wonder if the `else` clause where we `throw` is/was needed at all?